### ISSUE TITLE
Adding Changesets

### ISSUE THREAD

**Issue by AdrianGonz97** [skeletonlabs/skeleton#1519]
### Describe what feature you'd like. Pseudo-code, mockups, or screenshots of similar solutions are encouraged!

I'm going to try to layout all of my thoughts so that we're all on the same page on some of the technical reasonings behind some of these ideas.

We're still very much in the **experimental/exploratory/planning phase** of Changesets, so nothing has been implemented and definitely not set in stone. It's still going to take a bit of time to get everything in order and make things completely fool-proof - that's when we'll be satisfied enough to start opening PRs and begin implementation.

### Approach 1: Maintaining a `master` and `dev` branch on a time-based release cycle
The idea here is that we have a `dev` branch that we'll merge all of our PRs into and a `master` branch that represents the latest release of all of our packages. When a new PR is merged into `dev`, a new "dev/preview/nightly" package will be built and published under the `dev` dist-tag on npm. This will provide users an easy way to install and test out all of the latest features just by running `npm i @skeletonlabs/skeleton@dev` in their project.

#### PROS
- Maintains the our current 2 week release cycle
- When PRs are merged into `dev`, the package will be built and published with the `dev` dist-tag. Users will be able to install the package by adding it into the project like so: `npm i @skeletonlabs/skeleton@dev`.

#### CONS
- We *have to make sure* that the `master` and `dev` branches are _always_ in-sync with each other after releases. That means that when we merge `dev` into `master`, we *have* to select the *"merge with all commits"* option instead of squashing or rebasing. If we don't, and the two branches are out of sync of each other directly after a release, things are going to get messy in future releases if we don't correct it before the next one. (i.e. changesets that were applied in a previous release won't be deleted and will double up and apply again in the next release). Essentially, **we have the be _very_ careful with _how_ we merge `dev` into `master`**.
    - Perhaps there's a way to always sync them after releases with GH Actions? I'm not aware of a way how, but maybe some judo magic with git is required to do so, avoiding this issue altogether. Maybe deleting `dev` and re-branching from `master` after every release?
- This approach will also block the release of other packages **if one of them needs an update to be published immediately**. For example, say that we have a couple of updates already merged and ready to go on the `dev` branch and they're just waiting until the end of the release cycle to be merged into `master`. Now say that we need to push out and release an update to `create-skeleton-app` immediately. We won't be able to do it because we can't publish a package without affecting the other packages that have pending changesets. If there are any pending changesets, **they will be published together**.
    - NOTE: it _may_ be possible to configure GH Actions to account for this scenario, but it's going to **greatly** complicate the build scripts and our github action workflows, so I'm **really** not in favor of it
- We can't have `create-skeleton-app` in the monorepo due to the reason above.
 
---

### Approach 2: Maintaining just a `master` branch and publish a new release whenever a PR is merged
In this approach, we will just have one branch to maintain - no syncing to others is required. Whenever a PR is merged into `master`, the package will be built and published with the `latest` dist-tag on npm. Users will be able to use the latest features as soon as they are merged. This is approach is similar to how SvelteKit handles their releases.

#### PROS
- Very simple and easy to maintain CI/CD.
- Users will get the latest features as soon as they are merged.
- Publishing packages that need to be released immediately won't be blocked anymore.
- We'll be able to add `create-skeleton-app` to the monorepo.
- No need to worry about the small details on _how_ we merge into `master` since there's no syncing required.

#### CONS
- Not having a ongoing "dev/preview/nightly" release for testing newly **merged** PRs.
    - To make up for this, I've written a GH Action that will generate a preview release on PRs that contain a certain **label**. For example, say we that have a `test release` label for PRs. Once that label is added to a PR, the GH Action will run, building and publishing packages under the `dev` (we can also change it to `preview`) dist-tag. That way, we can still have a way to easily install the PR's package test features before they're officially merged into `master`.
- We have to be more thorough in what we merge into `master` making sure that they're ready.

---

In short, I believe the first approach is a lot more fragile and is more prone to errors. The simplicity of the second approach is certainly what makes it more fool-proof with less overall downsides.  

### What type of pull request would this be?

Enhancement

**Comment by niktek** [comment]
This was an interesting read: https://github.com/storybookjs/storybook/issues/22242 - the only problem with the new branch per release is that it is going to make it harder for non core contributors to send stuff in.
